### PR TITLE
Return typed dicts in client functions

### PR DESCRIPTION
**Pull request by gmargaritis**

### CODE REVIEW

File: ergani/client.py [hunk 1]
```diff
@@ -66,17 +98,52 @@ def _handle_response(
...
+
+        if not response:
+            return []
+
+        data = response.json()
+        submissions = []
+
+        for submission in data:
+            submission_date_str = submission["submitDate"]
+            submission_date = datetime.strptime(submission_date_str, "%d/%m/%Y %H:%M")
```
Comment (gmargaritis) [inline ergani/client.py:125]: Decided it would be best if parsed the string here and returned a proper datetime

Comment (parisk) [reply ergani/client.py:125]: Agreed.

File: ergani/models.py [hunk 2]
```diff
@@ -372,3 +372,17 @@ def serialize(self):
...
+
+        Attributes:
+            submission_id (str): The unique identifier of the submission
+            protocol (str): The protocol associated with the submission
+            submission_date (datetime): The datetime of the submission
+        """
+
+        submission_id: str
+        protocol: str
+        submission_date: datetime
```
[the remark this answers is not in the document]
Comment (parisk) [reply ergani/models.py:388]: Agreed.

**Review by parisk — APPROVED** [review]
LGTM 🦔

**Pull request merged**